components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/RoleManagementServiceImpl.java

```diff
@@ -0,0 +1,198 @@
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        return roleDAO.getRoles(filter, limit, offset, sortBy, sortOrder, tenantDomain);
+    }
+
+    @Override
+    public Role getRole(String roleName, String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        Role role = roleDAO.getRole(roleName, tenantDomain);
+        role.setPermissions(getPermissionListOfRole(roleName, tenantDomain));
+        return role;
+    }
+
+    @Override
+    public void updateRoleName(String roleName, String newRoleName, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        roleDAO.updateRoleName(roleName, newRoleName, tenantDomain);
+    }
+
+    @Override
+    public void deleteRole(String roleName, String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        roleDAO.deleteRole(roleName, tenantDomain);
+    }
+
+    @Override
+    public List<String> getUserListOfRole(String roleName, String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        return roleDAO.getUserListOfRole(roleName, tenantDomain);
+    }
+
+    @Override
+    public void updateUserListOfRole(String roleName, List<String> newUserList, List<String> deletedUserList,
+            String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        roleDAO.addUsersToRole(roleName, newUserList, tenantDomain);
+        roleDAO.removeUsersFromRole(roleName, deletedUserList, tenantDomain);
+    }
+
+    @Override
+    public List<String> getGroupListOfRole(String roleName, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        return roleDAO.getGroupListOfRole(roleName, tenantDomain);
+    }
+
+    @Override
+    public void updateGroupListOfRole(String roleName, List<String> newGroupNamesList,
+            List<String> deletedGroupNamesList, String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        roleDAO.addGroupsToRole(roleName, newGroupNamesList, tenantDomain);
+        roleDAO.removeGroupsFromRole(roleName, newGroupNamesList, tenantDomain);
+    }
+
+    @Override
+    public List<String> getPermissionListOfRole(String roleName, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        try {
+            return getSelectedPermissions(getUserAdminProxy().getRolePermissions(roleName, tenantId));
+        } catch (UserAdminException e) {
+            throw new IdentityRoleManagementException(
+                    "An error occurred when retrieving permissions of role : " + roleName, e);
```

Comment:
its better if we can pass error codes as well.

Reply:
Fixed and updated the PR.